lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +252,23 @@ def update_state_file
     end
   end
 
+  private
+
+  def enable_encoding
+    !@charset.nil? || !@columns_charset.empty?
+  end
+
+  # make sure the encoding is uniform over fields
+  def convert(column_name, value)
+    return value unless value.is_a?(String)
+    if @columns_charset[column_name]
+      converter = @converters[@columns_charset[column_name]]
+      converter.convert(value)
+    elsif @charset
+      converter = @converters[@charset]
+      converter.convert(value)
+    else
+      value
+    end
+  end
```

Comment:
I agree this code is more compact, but I dislike and thing is error prone to do things like initializations in the conditions like you propose in ``` if (column_name_charset = @columns_charset[column_name]) ```

Can we agree on middle term here and get this in? we can always refactor later on.